Example for training CNN with mixed precision

tlkh opened the pull request:
I've added an example for training a CNN to classify images from CIFAR10 using the new mixed-precision API in TF 1.14. 

This notebook explains briefly the idea of mixed-precision, and proceeds to train a CNN with and without mixed-precision to demonstrate the speed-up in training without sacrificing model performance.

The "Link to Colab" in the notebook is set to the link if the notebook is merged (I assumed I was supposed to do that, else the link needs to be edited after merging).

For review purposes, here is the [same notebook from my fork](https://colab.research.google.com/github/tlkh/tensorflow-examples/blob/master/community/en/cifar10_mixed_precision.ipynb).

This is my first contribution, please feel free to provide feedback or comments!

EDIT: identified some performance issues that can significantly improve performance, forced pushed over the previous commit.

margaretmz commented:
@tlkh very impressive PR! Suggestion: would be great to include some links to blog posts as background info in your notebook to explain what is **mixed precision** - for example, [Automatic Mixed Precision in TensorFlow for Faster AI Training on NVIDIA GPUs](https://medium.com/tensorflow/automatic-mixed-precision-in-tensorflow-for-faster-ai-training-on-nvidia-gpus-6033234b2540), and [Automatic Mixed Precision for NVIDIA Tensor Core Architecture in TensorFlow](https://devblogs.nvidia.com/nvidia-automatic-mixed-precision-tensorflow/)

tlkh commented:
Noted, I’ll be making some improvements then I’ll amend the commit

tlkh commented:
I have updated the commit and pushed again.

Minor changes to the rest of the notebook, but wrote a bit more about mixed-precision and added some links to useful resources.
